cmake/modules/AddSwift.cmake

```
@@ -958,7 +958,7 @@ endfunction()
 
 macro(add_swift_tool_symlink name dest component)
   add_llvm_tool_symlink(${name} ${dest} ALWAYS_GENERATE)
-  llvm_install_symlink(${name} ${dest} ALWAYS_GENERATE COMPONENT ${component})
+  llvm_install_symlink(LLVM ${name} ${dest} ALWAYS_GENERATE COMPONENT ${component})
```

The first parameter is the project, I think that this should be `SWIFT`, not `LLVM`.

```suggestion
  llvm_install_symlink(SWIFT ${name} ${dest} ALWAYS_GENERATE COMPONENT ${component})
```

---

I used `LLVM` since from a `trace-expand` run it seemed `SWIFT` may not have worked, and I had a desire to unblock the build of `next` with a simple change (although I recognize it may not be the right call).

To expand a bit on my findings -- the `project` parameter is used to infer the output directory for the symlink

```
 set(output_dir "${${project}_TOOLS_INSTALL_DIR}")

  install(SCRIPT ${INSTALL_SYMLINK}
          CODE "install_symlink(\"${full_name}\" \"${full_dest}\" \"${output_dir}\")"
          COMPONENT ${component})
```

https://github.com/apple/llvm-project/blob/a425ae53badef513e660538b8c18a024d977f48d/llvm/cmake/modules/AddLLVM.cmake#L2082

and I could not see a `SWIFT_TOOLS_INSTALL_DIR` defined anywhere -- I could have added one, but I was afraid of making too big of chance with the little understanding I had.

Also this seems to match the implementations of the two functions called by `add_swift_tool_symlink ` in `main` -- both hardcoded `LLVM` as the project name.

---

https://github.com/llvm/llvm-project/search?q=llvm_install_symlink shows that each project is using their own identity.

---

Thanks for pointing that out, I realized only now my haste was not justified.

Incorporating this feedback in 79b5a1d631635a7a226fda7a7b6dd53644e177ff, where I defined `SWIFT_TOOLS_INSTALL_DIR` to correctly generate the installation script and I use `llvm_add_tool_symlink` instead of `add_llvm_tool_symlink` for uniformity